pandas/rpy/common.py

```diff
@@ -134,35 +137,47 @@ def _convert_str_vector(obj):
     return arr
 
 
-def _convert_DataFrame(rdf):
+def _convert_factor(obj, **kwargs):
+    if kwargs.get("factors_as_strings", True):
+        levels = np.asarray(obj.levels)
+        values = np.asarray(obj)
+        if com.is_float_dtype(values):
+            mask = np.isnan(values)
```

Comment:
@jseabold: Sorry, my bad. I see now that that code was added due to
GH #1615. Unfortunately, there appears to be a regression when
`factors_as_strings=False`:

```
from pandas.rpy.common import load_data
prestige = load_data('Prestige', 'car', factors_as_strings=False)
``` 

raises `ValueError: codes need to be between -1 and len(categories)-1`.

And the issue also affects `factors_as_strings=True`, where

```
prestige = load_data('Prestige', 'car', factors_as_strings=True)
```

raises `IndexError: index 2147483647 is out of bounds for axis 0 with size 3`